src/crimson/os/seastore/segment_manager/block.cc

```diff
@@ -501,11 +519,87 @@ SegmentManager::read_ertr::future<> BlockSegmentManager::read(
     return crimson::ct_error::enoent::make();
   }
 
+  stats.data_read.increment(len);
   return do_read(
     device,
     get_offset(addr),
     len,
     out);
 }
 
+void BlockSegmentManager::register_metrics()
+{
+  namespace sm = seastar::metrics;
+  // TODO: add label for device_id
+  auto counter_label = sm::label("counter");
+  auto num_label_instance = counter_label("NUM");
+  auto bytes_label_instance = counter_label("BYTES");
+  stats.reset();
+  metrics.add_group(
+    "segment_manager",
+    {
+      sm::make_counter(
+        "data_read",
+        stats.data_read.num,
+        sm::description("total data read effort"),
+        {num_label_instance}
```

Comment:
i don't think if this is the best way to use the "label" of metrics. namely, to name inherently different metrics with the same name but different labels.

instead of using "label" as the name of properties of a metric, i think the idea of "label" is to differentiate one metric from another with the same name. the difference between them is just they belong to different group or entity. for instance, IMHO, a label can used to categorize latencies at different phases. or it can be used to differentiate the cache hit ratio of two different storage device. but, to use it to name different metrics.. i'd say it would be confusing and is not consistent to how seastar and we use it.

so, i'd suggest use different names for these metrics, for example, 

- "data_read_exts" for the number of extents which have been read as data,
- "data_read_bytes" for the total size of extents which have been read as data

Reply:
Yeah, it makes sense to me.

Reply:
"data_read_exts" is better named "data_read_count", as it counts how many reads are done at segment manager layer, regardless of extents.

Reply:
Fixed, I'm working on another PR to refine cache layer metrics.